API.md

```diff
@@ -0,0 +1,213 @@
+<!-- EDIT AT https://raw.githubusercontent.com/TheThingsNetwork/node-app-lib/master/API.md -->
+
+# API Reference
+
+Require the TTN Client module:
+
+```java
+import org.thethingsnetwork.java.app.lib.Client;
+import org.thethingsnetwork.java.app.lib.events.*;
+```
+
+## Class: Client
+
+Creates an instance of the client:
+
+```java
+Client client = new Client(region, appId, accessKey [, connOpts]);
+```
+
+* `region [String]`: The region (e.g. `eu`) or full hostname (e.g. `eu.thethings.network`) of the handler to connect to.
+* `appId [String]`: The ID of the application to connect to (e.g. `hello-world`).
+* `appAccessKey [String]`: An access key for the application, formatted as base64 (e.g. `'2Z+MU0T5xZCaqsD0bPqOhzA6iygGFoi4FAgMFgBfXSo='`).
+* `connOpts [MqttConnectOptions]`: Some custom configuration of the MQTT connection. This parameter is optional.
+
+## Event: connect
+
+Emitted on successful connection.
+
+```java
+client.onConnected(new ConnectHandler() {
+    public void handle(MqttClient client) {
+        System.out.println("connected !");
+    }
+});
+```
+
+* `client [MqttClient]`: MQTT connection wrapper. See [MQTT](http://www.eclipse.org/paho/files/javadoc/org/eclipse/paho/client/mqttv3/MqttClient.html).
+
+## Event: error
+
+Emitted when the client cannot connect or when a parsing error occurs.
+
+```java
+client.onError(new ErrorHandler() {
+    public void handle(Throwable error) {
+        System.err.println("error: " + error.getMessage());
+    }
+});
+```
+
+* `error [Throwable]`: Error object. See [MQTT](https://docs.oracle.com/javase/8/docs/api/java/lang/Exception.html).
+
+## Event: message
+
+Emitted when TTN forwards a message addressed to your application.
+
+```java
+client.onUplink(new UplinkHandler() {
+    public void handle(String devId, Object data) {
+        System.out.println("Message: " + devId + " " + data);
+    }
+});
+```
+
+* `devId [String]`: Device ID, e.g.: `my-uno`.
+* `data [Object]`: Message data, e.g.:
+
+  ```json
+  {
+    "port": 1,
+    "counter": 10,
+    "payload_raw": "MQ==",
+    "payload_fields": {
+      "led": true
+    },
+    "metadata": {
+      "time": "2016-09-07T12:50:07.068771281Z",
+      "frequency": 868.1,
+      "modulation": "LORA",
+      "data_rate": "SF7BW125",
+      "coding_rate": "4/5",
+      "gateways": [{
+        "gtw_id": "0000024B08060112",
+        "timestamp": 3955426155,
+        "time": "2016-09-07T12:50:07.053048Z",
+        "channel": 4,
+        "rssi": -109,
+        "snr": 5.8,
+        "rf_chain": 1
+      }]
+    }
+  }
+  ```
+
+### Listen for a specific device
+
+```java
+client.onUplink(new UplinkHandler() {
+    public void handle(String devId, Object data) {
+        System.out.println("Message: " + devId + " " + data);
+    }
+    public String getDevId(){
```

Comment:
Why does this need to be a function returning the string? Can't it just be a string, ideally even as first argument?